src/Sulu/Bundle/Sales/OrderBundle/Controller/PdfController.php

```diff
@@ -61,16 +60,62 @@ public function orderConfirmationAction(Request $request, $id)
 
         $pdf = $this->getPdfManager()->createOrderConfirmation($orderApiEntity);
 
-        $pdfName = $this->getPdfManager()->getPdfName($order);
+        // Get pdf name with function parameter isSubmitted = true.
+        $pdfName = $this->getPdfManager()->getPdfName($order, true);
         $responseType = $this->container->getParameter('sulu_sales_order.pdf_response_type');
 
         return new Response(
             $pdf,
             200,
-            array(
+            [
                 'Content-Type' => 'application/pdf',
                 'Content-Disposition' => $responseType . '; ' . 'filename="' . $pdfName . '"'
-            )
+            ]
+        );
+    }
+
+    /**
+     * Finds an order object by a given id from the url and renders a configurable template.
+     * Then returns it as pdf in a new tab.
+     *
+     * @param Request $request
+     * @param int $id
+     *
+     * @throws OrderNotFoundException
+     *
+     * @return Response
+     */
+    public function orderPdfAction(Request $request, $id)
+    {
+        $locale = $this->getLocale($request);
+
+        $this->get('translator')->setLocale($locale);
+
+        $orderRepository = $this->get('sulu_sales_order.order_repository');
+
+        try {
+            $orderApiEntity = $this->getOrderManager()->findByIdAndLocale($id, $locale);
+            $order = $orderApiEntity->getEntity();
+        } catch (OrderNotFoundException $exc) {
+            throw new OrderNotFoundException($id);
```

Comment:
actually you don't need to catch the exception, since you are just throwing it again.